webpush/encrypt.go

```diff
@@ -144,16 +170,25 @@ type EncryptionResult struct {
 //    - https://tools.ietf.org/html/draft-ietf-httpbis-encryption-encoding
 //    - https://en.wikipedia.org/wiki/Elliptic_curve_Diffie%E2%80%93Hellman
 //    - https://tools.ietf.org/html/draft-ietf-webpush-encryption
-func Encrypt(sub *Subscription, message string) (*EncryptionResult, error) {
+func Encrypt(sub *Subscription, message string, encoding ContentEncoding) (*EncryptionResult, error) {
 	plaintext := []byte(message)
+
+	var maxPayloadLength int
+	if encoding == AESGCM {
+		maxPayloadLength = aesgcmMaxPayloadLength
+	} else {
+		maxPayloadLength = aes128gcmMaxPayloadLength
+	}
 	if len(plaintext) > maxPayloadLength {
 		return nil, fmt.Errorf("Payload is too large. The max number of bytes is %d, input is %d bytes.", maxPayloadLength, len(plaintext))
 	}
 
+	// sub.Key is the p256dh key.
 	if len(sub.Key) == 0 {
```

Comment:
You could check this and sub.Auth at the top of the function, before anything else, to avoid unnecessary allocations. It would also make the main flow more readable